### DISCUSSION TITLE
Checkbox should accept boolean for value prop

### DISCUSSION THREAD

**Discussion by mcmxcdev** [mantinedev/mantine#3076] [Feature requests]
Currently, `Checkbox` component accept value like this: `value?: string | number | readonly string[] | undefined`

A lot of times though, a single `Checkbox` is used without a `CheckboxGroup` like on the default usage example: https://mantine.dev/core/checkbox/#usage 

In our case, we connect up a form with `react-hook-form` containing this checkbox. Therefore, we would expect to be able to pass in a true/false value for the controlled input.

**Comment by rtivital** [comment]
Mantine does not add any extra logic on top of `input[type="checkbox"]` – its value cannot be boolean. You should set `checked` prop instead.

**Reply by erfan-goodarzi** [reply]
Actually there is problem here. we have a situation that use `CheckboxGroup`. but also we want to access the checked props in `Checkbox`. Whenever we use `Checkbox` with `CheckboxGroup` the checked props wont work. do u have any idea?

**Reply by rtivital** [reply]
Checkbox state in checkbox group is stored in the array. To get checked state, use `checkboxGroupValues.includes('checkbox-value')`.

**Reply by erfan-goodarzi** [reply]
hmmm i couldn't figure it out.
So this is my situation:

![carbon](https://github.com/mantinedev/mantine/assets/63591256/2f63bfdc-fc98-4b13-b526-0c781605dbcd)


Now how can i do what u said?
